src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs

```diff
@@ -583,5 +618,79 @@ private string[] GetCompatibilityLevels(int sqlServerVersion, DatabasePrototype
+        {
+            var levels = new List<AzureEditionDetails>();
+            foreach (string displayName in editionDisplayNames)
+            {
+                if (AzureSqlDbHelper.TryGetAzureEditionFromDisplayName(displayName, out var azureEdition))
+                {
+                    if (AzureSqlDbHelper.TryGetServiceObjectiveInfo(azureEdition, out var serviceInfoPair))
+                    {
+                        // Move default value to the front of the list
+                        var serviceLevelsList = new List<string>(serviceInfoPair.Value);
+                        var defaultIndex = serviceInfoPair.Key;
+                        if (defaultIndex >= 0 && defaultIndex < serviceLevelsList.Count)
+                        {
+                            var defaultServiceObjective = serviceLevelsList[defaultIndex];
+                            serviceLevelsList.RemoveAt(defaultIndex);
+                            serviceLevelsList.Insert(0, defaultServiceObjective);
+                        }
+                        var details = new AzureEditionDetails() { EditionDisplayName = displayName, Details = serviceLevelsList.ToArray() };
+                        levels.Add(details);
+                    }
+                }
+            }
+            return levels.ToArray();
+        }
+
+        /// <summary>
+        /// Get supported editions for this Azure server.
+        /// </summary>
+        private static string[] GetAzureEditions()
+        {
+            return AzureSqlDbHelper.GetValidAzureEditionOptions().Select(edition => edition.DisplayName).ToArray();
+        }
+
+        /// <summary>
+        /// Get supported maximum sizes for this Azure server.
+        /// </summary>
+        private static AzureEditionDetails[] GetAzureMaxSizes(string[] editionDisplayNames)
+        {
+            var sizes = new List<AzureEditionDetails>();
+            foreach (string displayName in editionDisplayNames)
+            {
+                if (AzureSqlDbHelper.TryGetAzureEditionFromDisplayName(displayName, out var azureEdition))
+                {
+                    if (AzureSqlDbHelper.TryGetDatabaseSizeInfo(azureEdition, out var sizeInfoPair))
+                    {
+                        // Move default value to the front of the list
+                        var sizeInfoList = new List<DbSize>(sizeInfoPair.Value);
+                        var defaultIndex = sizeInfoPair.Key;
+                        if (defaultIndex >= 0 && defaultIndex < sizeInfoList.Count)
+                        {
+                            var defaultSizeInfo = sizeInfoList[defaultIndex];
+                            sizeInfoList.RemoveAt(defaultIndex);
+                            sizeInfoList.Insert(0, defaultSizeInfo);
+                        }
+                        var details = new AzureEditionDetails() { EditionDisplayName = displayName, Details = sizeInfoList.Select(info => info.ToString()).ToArray() };
```

Comment:
Is there a reason this stuff is being calculated dynamically each time? The edition -> available size mappings seems like another value that could be made static.

Reply:
This is the builder method that gets called from the static constructor, so it only gets called once.